### PR TITLE
fix: workspace title editing

### PR DESCRIPTION
**Pull request by danielchim**
meeting the lastest design spec from #2645 for workspace editing.

### CODE REVIEW

File: apps/core/src/components/blocksuite/workspace-header/index.tsx [hunk 1]
```diff
@@ -56,50 +55,48 @@ export const BlockSuiteEditorHeader: FC<
...
-              <EditorModeSwitch
-                blockSuiteWorkspace={workspace.blockSuiteWorkspace}
-                pageId={currentPage.id}
-                style={{
-                  marginRight: '12px',
-                }}
-              />
-            </div>
+            {isEditable ? (
+              <></>
```
Comment (himself65) [inline apps/core/src/components/blocksuite/workspace-header/index.tsx]: null

File: apps/core/src/components/blocksuite/workspace-header/index.tsx [hunk 2]
```diff
@@ -56,50 +55,48 @@ export const BlockSuiteEditorHeader: FC<
...
             </div>
-            <div className={styles.searchArrowWrapper}>
-              <QuickSearchButton
-                onClick={() => {
-                  setOpenQuickSearch(true);
-                }}
-              />
-            </div>
+            {isEditable ? (
+              <></>
```
Comment (himself65) [inline apps/core/src/components/blocksuite/workspace-header/index.tsx]: null?

File: tests/affine-local/e2e/change-page-mode.spec.ts [hunk 3]
```diff
@@ -70,7 +70,7 @@ test('Able to insert the title of an untitled page', async ({ page }) => {
   await titleBarTextContent.click({ delay: 100 });
   const titleContent = await page.getByTestId('title-content');
   await titleContent.fill('test');
-  await page.getByTestId('save-edit-button').click({ delay: 100 });
+  await page.mouse.click(600, 600, { delay: 100 });
```
Comment (himself65) [inline tests/affine-local/e2e/change-page-mode.spec.ts]: don't use fixed position

**Review by himself65 — CHANGES_REQUESTED** [review]

**Comment by JimmFly** [comment]
Looking forward to your PR being merged.

**Pull request closed without merging**